api/src/DTO/ResetPassword.php

```diff
@@ -2,38 +2,43 @@
 
 namespace App\DTO;
 
-use ApiPlatform\Core\Annotation\ApiProperty;
-use ApiPlatform\Core\Annotation\ApiResource;
+use ApiPlatform\Metadata\ApiProperty;
+use ApiPlatform\Metadata\ApiResource;
+use ApiPlatform\Metadata\Get;
+use ApiPlatform\Metadata\Patch;
+use ApiPlatform\Metadata\Post;
+use App\State\ResetPasswordCreateProcessor;
+use App\State\ResetPasswordProvider;
+use App\State\ResetPasswordUpdateProcessor;
 use Symfony\Component\Serializer\Annotation\Groups;
 use Symfony\Component\Validator\Constraints as Assert;
 
 #[ApiResource(
-    collectionOperations: [
-        'post' => [
-            'security' => 'true',
-            'path' => '',
-            'status' => 204,
-            'output' => false,
-            'denormalization_context' => ['groups' => ['create']],
-            'normalization_context' => ['groups' => ['read']],
-            'openapi_context' => [
-                'summary' => 'Request Password-Reset-Mail',
-                'description' => 'Password-Reset-Link will be sent to the given email',
-            ],
-        ],
+    operations: [
+        new Get(
+            provider: ResetPasswordProvider::class,
+            uriTemplate: '/reset_password/{id}{._format}', // TO DISCUSS: default uri would be /reset_password (plural). Shall we keep or fix?
+            security: 'true',
+        ),
+        new Patch(
+            provider: ResetPasswordProvider::class,
+            processor: ResetPasswordUpdateProcessor::class,
+            uriTemplate: '/reset_password/{id}{._format}',
+            security: 'true',
+            denormalizationContext: ['groups' => ['update']]
+        ),
+        new Post(
+            processor: ResetPasswordCreateProcessor::class,
+            uriTemplate: '/reset_password{._format}',
+            security: 'true',
+            status: 204,
+            output: false,
+            denormalizationContext: ['groups' => ['create']],
+            normalizationContext: ['groups' => ['read']],
+            openapiContext: ['summary' => 'Request Password-Reset-Mail', 'description' => 'Password-Reset-Link will be sent to the given email']
+        ),
     ],
-    itemOperations: [
-        'get' => [
-            'security' => 'true',
-            'path' => '/{id}',
-        ],
-        'patch' => [
-            'security' => 'true',
-            'path' => '/{id}',
-            'denormalization_context' => ['groups' => ['update']],
-        ],
-    ],
-    routePrefix: '/auth/reset_password'
+    routePrefix: '/auth'
```

Comment:
Originally because of https://github.com/api-platform/core/issues/5203

But I would still keep it as it is now even after https://github.com/api-platform/core/issues/5203 is fixed. This is more consistent with the auto-generated operations, where `uriTemplate` starts with the resource name and `routePrefix` is only used for additional namespacing.